hatch_pip_compile/installer.py

```diff
@@ -0,0 +1,149 @@
+        """
+        with self.environment.safe_activation():
+            self.environment.platform.check_command(
+                self.environment.construct_pip_install_command(
+                    args=["--no-deps", str(self.environment.root)]
+                )
+            )
+
+    def install_project_dev_mode(self) -> None:
+        """
+        Install the project in editable mode (`--no-deps`)
+        """
+        with self.environment.safe_activation():
+            self.environment.platform.check_command(
+                self.environment.construct_pip_install_command(
+                    args=["--no-deps", "--editable", str(self.environment.root)]
+                )
+            )
+
+
+class PipInstaller(PluginInstaller):
+    """
+    Plugin Installer for `pip`
+    """
+
+    def install_dependencies(self) -> None:
+        """
+        Install the dependencies with `pip`
+        """
+        with self.environment.safe_activation():
+            if not self.environment.piptools_lock_file.exists():
+                return
+            extra_args = self.environment.config.get("pip-compile-install-args", [])
+            args = [*extra_args, "--requirement", str(self.environment.piptools_lock_file)]
+            install_command = self.environment.construct_pip_install_command(args=args)
+            self.environment.virtual_env.platform.check_command(install_command)
+
+
+class PipSyncInstaller(PluginInstaller):
+    """
+    Plugin Installer for `pip-sync`
+    """
+
+    def install_dependencies(self) -> None:
+        """
+        Install the dependencies with `pip-sync`
+
+        In the event that there are no dependencies, pip-sync will
+        uninstall everything in the environment before deleting the
+        lockfile.
+        """
+        self.environment.install_pip_tools()
+        cmd = [
+            self.environment.virtual_env.python_info.executable,
+            "-m",
+            "piptools",
+            "sync",
+            "--verbose"
+            if self.environment.config.get("pip-compile-verbose", None) is True
+            else "--quiet",
+            "--python-executable",
+            str(self.environment.virtual_env.python_info.executable),
+        ]
+        if not self.environment.dependencies:
+            self.environment.piptools_lock_file.write_text("")
+        extra_args = self.environment.config.get("pip-compile-install-args", [])
+        cmd.extend(extra_args)
+        cmd.append(str(self.environment.piptools_lock_file))
+        self.environment.virtual_env.platform.check_command(cmd)
+        if not self.environment.dependencies:
+            self.environment.piptools_lock_file.unlink()
+
+    def _full_install(self) -> None:
+        """
+        Run the full install process
+
+        1) Run pip-compile (if necessary)
+        2) Run pip-sync
+        3) (re)install project
+        """
+        with self.environment.safe_activation():
+            self.install_dependencies()
```

Comment:
See https://github.com/juftin/hatch-pip-compile/pull/34#discussion_r1416565511

There is a scenario where the lockfile doesn't exist and `install_dev_mode` invokes `_full_install` without `run_pip_compile` ever being called

```suggestion
        with self.environment.safe_activation():
            self.environment.run_pip_compile()
            self.install_dependencies()
```